catris.py

```diff
@@ -568,15 +673,15 @@ def _send_queue_thread(self) -> None:
                     print(self.client_address, e)
 
             with self.server.access_game_state() as state:
-                if self in self.server.clients:
-                    self.server.clients.remove(self)
-                    if isinstance(
-                        self.player.moving_block_or_wait_counter, MovingBlock
-                    ):
-                        self.player.moving_block_or_wait_counter = None
-
-                    if not self.server.clients:
-                        state.reset()
+                self.server.clients.remove(self)
+                if isinstance(self.view, PlayingView) and isinstance(
+                    self.view.player.moving_block_or_wait_counter, MovingBlock
+                ):
+                    self.view.player.moving_block_or_wait_counter = None
+                if not any(
+                    isinstance(c.view, PlayingView) for c in self.server.clients
+                ):
+                    state.reset()
```

Comment:
It can reset many times, when everyone are in game over view and quit one by one. It just doesn't feel right to have this here.